src/query.rs

```diff
@@ -1082,6 +1084,91 @@ mod tests {
             so they won't be part of cargo-semver-checks: {missing_lints:?}"
         )
     }
+
+    #[test]
+    fn test_data_is_fresh() -> anyhow::Result<()> {
+        // Adds the modification time of all files in `{dir}/**/*` to `set`
+        fn recursive_file_times<P: Into<PathBuf>>(
+            dir: P,
+            set: &mut BTreeSet<SystemTime>,
+        ) -> std::io::Result<()> {
+            for item in fs_err::read_dir(dir)? {
+                let item = item?;
+                let metadata = item.metadata()?;
+                if item.file_name() == "target" || item.file_name() == "Cargo.lock" {
+                    continue;
+                }
```

Comment:
I agree with dropping `target` from the recursion and skipping the lockfile. But perhaps we should actively only select `.rs` or `.toml` files in addition?

What I'm worried about is something silly like "macOS made a new `.DS_Store` file in the directory" causing tests to fail. I don't think we can hope to exclude all such possible ephemeral files on all operating systems. If we fail to exclude such a fail, we get a false-positive test result here: the test suite would + should have passed, but this test prevented that.

I think an explicit "allow list" of files we check is better, and a rule like "don't recurse into `target`" plus "check times for `*.rs` and `*.toml` files" should be sufficient.